turbopack/crates/turbopack-core/src/source_map/fixed_source_map_asset.rs

```diff
@@ -0,0 +1,94 @@
+use anyhow::Result;
+use turbo_rcstr::RcStr;
+use turbo_tasks::{FxIndexSet, ResolvedVc, ValueToString, Vc};
+use turbo_tasks_fs::{File, FileSystemPath};
+
+use crate::{
+    asset::{Asset, AssetContent},
+    introspect::{Introspectable, IntrospectableChildren},
+    output::OutputAsset,
+    source_map::{GenerateSourceMap, SourceMap},
+};
+
+/// Represents the source map of an ecmascript asset.
+#[turbo_tasks::value]
+pub struct FixedSourceMapAsset {
```

Comment:
Could this just be a different `new` constructor on the existing `SourceMapAsset`?